test/spec/request-test.js

```diff
@@ -1425,4 +1427,263 @@ describe('Request-Promise', function () {
 
     });
 
+
+    describe('should implement a HTTP 1.1 consistent cache behavior', function () {
+        var cacheServer;
+        var now = moment().format(LAST_MODIFIED_FORMAT)
+
+        var resources = [
+            {
+                header: {
+                    'Last-Modified': now
+                    'Cache-Control': 'public, max-age=10'
+                    'Content-Type': 'application/json'
+                }
+                content: {
+                    name: 'resource1'
+                }
+            },
+            {
+                header: {
+                    'Last-Modified': now
+                    'Content-Type': 'application/json'
+                }
+                content: {
+                    name: 'resource2'
+                }
+            },
+            {
+                header: {
+                    'Last-Modified': now
+                    'Cache-Control': 'no-cache'
+                    'Content-Type': 'application/json'
+                }
+                content: {
+                    name: 'resource3'
+                }
+            },
+            {
+                header: {
+                    'Last-Modified': now
+                    'Cache-Control': 'no-cache'
+                    'Content-Type': 'application/json'
+                    'Age': 20
+                }
+                content: {
+                    name: 'resource4'
+                }
+            }
+        ]
+
+        var incrementResource = function(resource) {
+            resource.header['Last-Modified'].add(5, 'seconds')
+        }
```

Comment:
this method simulates a change of the resource (update in db that would for instance increase the last modified information)